site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+
+In this tutorial, we will start with an application that is managing its state in the widgets themselves. We will then extract all of the state-related code out of the widgets and inject external state as properties only into widgets as is needed. You can [download](../assets/1010_form_widgets-initial.zip) the demo project to get started.
+
+## Prerequisites
+
+This tutorial assumes that you have gone through the [beginner tutorial](../001_static_content) series.
+
+{% section %}
+
+## Creating an application context
+
+{% task 'Create a class to manage application state.' %}
+
+To begin our discussion, let's review the initial version of `App`:
+
+{% include_codefile 'demo/initial/biz-e-corp/src/widgets/App.ts' %}
+
+Most of this widget is dedicated to holding and managing the `WorkerData` in the application. Notice, however, that it never actually uses that data itself. `App` is only containing the state and passing it to the children as required via properties. Up lifting state to the highest common widget in the tree is a valid pattern but as an application's state grows in size and complexity it is often desirable to decouple this from widgets. In larger applications, the `App` class would become complicated and more difficult to maintain due to the additional state information that it would be required track. Since the state information is not a primary concern of the `App` class, let's refactor it out of `App` and into a new `ApplicationContext` class.
+
+{% instruction 'Add the following to `ApplicationContext.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' %}
+
+The code begins by importing some modules, including the `WorkerProperties` and `WorkerFormData` interfaces defined in the `Worker` and `WorkerForm` modules. These two interfaces define the shape of state that the `ApplicationContext` manages.
+
+The `ApplicationContext` class contains the state information. `ApplicationContext` has two private fields, `_workerData` and `_formData`, which contain the state, and two accessor methods to retrieve these fields. Finally, notice the two public methods that `ApplicationContext` exposes:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' lines:26-35 %}
+
+The `formInput` method provides the same functionality as the `_onFormInput` method in the `App` class and the `submitForm` method is analogous to `_addWorker` from the `App` class. The implementations vary slightly as the `ApplicationContext` has dedicated fields to store the state information. Also, since the `ApplicationContext` is not a widget, it cannot call `invalidate();` to schedule a re-render, the instance needs to emit an `invalidate` event that instructs associated widgets to `invalidate` themselves.
```

Comment:
"a re-render, the instance needs to emit" -> "a re-render. Instead the instance needs to emit"